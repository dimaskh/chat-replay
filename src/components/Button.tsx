// Libs
import React from 'react'

// Styles
import styled from 'styled-components'

type Props = {
  /** Button title string */
  title: string
  /** Function that handles onClick */
  onClick: () => void
}

export const Button: React.FC<Props> = ({ title = '', onClick }: Props) => (
  <StyledButton onClick={onClick}>{title}</StyledButton>
)

const StyledButton = styled.div`
  color: red;
  cursor: pointer;
  user-select: none;
  background-color: #f18844;
  color: black;
  width: 180px;
`

export default Button
