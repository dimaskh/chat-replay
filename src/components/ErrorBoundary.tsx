// Libs
import React from 'react'

// Styles
import styled from 'styled-components'

export default class ErrorBoundary extends React.PureComponent {
  public state = {
    error: null,
    info: null,
  }

  public componentDidCatch(error: any, info: any) {
    this.setState({ error, info })
  }

  public render() {
    const { error } = this.state

    if (error) {
      return <ErrorBoundaryFallbackComponent />
    }

    return this.props.children
  }
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`

const Message = styled.div`
  padding: 40px;
  border: 2px #78909c solid;
  border-radius: 5px;
  font-size: 24px;
  color: #78909c;
`

const ErrorBoundaryFallbackComponent = () => (
  <Container>
    <Message>
      Some Error Occurred
      <span role="img" aria-label="face-emoji">
        ðŸ˜ž
      </span>
    </Message>
  </Container>
)
