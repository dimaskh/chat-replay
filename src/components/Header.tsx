// Libs
import React from 'react'

// Utils
import styled from 'styled-components'

export type Props = {
  children: string
  medium?: boolean
  small?: boolean
}

export const Header: React.FC<Props> = ({ children, ...props }: Props) => (
  <StyledHeader {...props}>{children}</StyledHeader>
)

export const StyledHeader = styled.header`
  background-color: #282c34;
  height: ${(props: Props) => {
    if (props.medium) return '60px'
    if (props.small) return '40px'
    return '100px'
  }};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: ${(props: Props) => {
    if (props.medium) return '20px'
    if (props.small) return '12px'
    return '32px'
  }};
  color: white;
  text-align: center;
  border-bottom: 1px solid #737373;
  width: 100%;
`

export default Header
