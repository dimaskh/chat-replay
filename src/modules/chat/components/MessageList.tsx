import { MessageList } from 'react-chat-elements'

// Default export to be used in dynamic import
export default MessageList
