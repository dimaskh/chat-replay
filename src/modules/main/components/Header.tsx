// Libs
import React from 'react'

// Utils
import styled from 'styled-components'

export const Header: React.FC = () => (
  <StyledHeader>
    <p>Chat Replay Application</p>
  </StyledHeader>
)

const StyledHeader = styled.header`
  background-color: #282c34;
  height: 100px;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  color: white;
  text-align: center;
`

export default Header
