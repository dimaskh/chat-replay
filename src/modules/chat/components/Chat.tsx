// Libs
import React from 'react'

// Components
import Header from 'components/Header'
import { MessageList } from 'react-chat-elements'

// Styles
import styled from 'styled-components'

export class Chat extends React.PureComponent<any> {
  public render() {
    return (
      <StyledChat>
        <Header medium>Chat Replay</Header>
        {this.renderContent()}
      </StyledChat>
    )
  }

  private renderContent = (): JSX.Element => {
    const { messages } = this.props

    return <MessageList className="message-list" dataSource={messages} />
  }
}

export const StyledChat = styled.div`
  display: flex;
  flex-direction: column;
  width: 55vw;
  height: 100%;
  color: white;
`

export default Chat
