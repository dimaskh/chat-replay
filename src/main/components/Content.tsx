// Libs
import React from 'react'

// Components
import Chat from 'modules/chat/components/Chat'
import Player from 'modules/player/containers/Player'
import UsersList from 'modules/users/components/UsersList'

// Styles
import styled from 'styled-components'

export const Content: React.FC = () => (
  <StyledContent>
    <Player />
    <UsersList />
    <Chat />
  </StyledContent>
)

const StyledContent = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
  background-color: #282c34;
`

export default Content
