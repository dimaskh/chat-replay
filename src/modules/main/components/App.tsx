// Libs
import React from 'react'

// Components
import Header from 'components/Header'
import Content from './Content'

// Styles
import styled from 'styled-components'

export const App: React.FC = () => (
  <StyledApp>
    <Header>Chat Replay Application</Header>
    <Content />
  </StyledApp>
)

const StyledApp = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
`

export default App
