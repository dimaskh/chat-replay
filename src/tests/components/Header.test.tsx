// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Tested components
import Header, { StyledHeader } from 'components/Header'

// Types
import { Props } from 'components/Header'
import { ShallowWrapper } from 'enzyme'

describe('components/Header', () => {
  let component: ShallowWrapper<Props>

  beforeEach(() => {
    component = shallow(<StyledHeader>Title</StyledHeader>)
  })

  it('should render Header correctly', () => {
    const HeaderComponent = shallow(<Header>Title</Header>)
    expect(shallowToJson(HeaderComponent)).toMatchSnapshot()
  })

  it('should render StyledHeader correctly', () => {
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should render StyledHeader correctly with medium prop', () => {
    component.setProps({ medium: true })
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should render StyledHeader correctly with small prop', () => {
    component.setProps({ medium: false, small: true })
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
