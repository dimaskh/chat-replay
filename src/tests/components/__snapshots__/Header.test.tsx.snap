// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header should render Header correctly 1`] = `
<styled.header>
  Title
</styled.header>
`;

exports[`components/Header should render StyledHeader correctly 1`] = `
<header
  className="sc-AxjAm iMRwGC"
>
  Title
</header>
`;

exports[`components/Header should render StyledHeader correctly with medium prop 1`] = `
<header
  className="sc-AxjAm hVedNm"
>
  Title
</header>
`;

exports[`components/Header should render StyledHeader correctly with small prop 1`] = `
<header
  className="sc-AxjAm iPBSjx"
>
  Title
</header>
`;
